packages/back-end/src/integrations/BigQuery.ts

```diff
@@ -78,4 +81,46 @@ export default class BigQuery extends SqlIntegration {
   castUserDateCol(column: string): string {
     return `CAST(${column} as DATETIME)`;
   }
+
+  async getInformationSchema(
+    dataSourceType: DataSourceType,
+    projectId: string
+  ): Promise<InformationSchema[] | null> {
+    if (!projectId) {
+      return null;
+    }
+
+    const datasets = await this.runQuery(
+      `SELECT * FROM ${projectId}.INFORMATION_SCHEMA.SCHEMATA;`
+    );
+
+    const combinedResults: RawInformationSchema[] = [];
+
+    const query: string[] = [];
+
+    for (const dataset of datasets) {
+      query.push(`SELECT
+        "${projectId}" as table_catalog,
+        table_name,
+        column_name,
+        data_type,
+        table_schema
+      FROM
+        ${projectId}.${dataset.schema_name}.INFORMATION_SCHEMA.COLUMNS`);
```

Comment:
Some data warehouses will have 100,000 rows returned for this query, so we can't query it all at once.  Instead, we should do 2 passes.

1. Do this query with a `GROUP BY` clause to get one row per table with `COUNT(*)` to get the number of columns
2. If the total number of columns across all tables is less than some threshold (500?), then re-run the query without a `GROUP BY`.

So for large data warehouses, we will only get table-level info back, which is fine for now.  In the future, we can build a way to incrementally build this index in the background and/or fetch columns for specific tables on demand as they are needed.

Reply:
@jdorn I've updated this based on our conversation yesterday. I wasn't able to get the `GROUP BY` to work as expected, but I was able to get it to work with `DISTINCT`.

Now, we're only getting databases, schemas, and a list of table names, and eventually calculating the number of columns per table. There are efficiencies to be had here if we can get the `COUNT` and `GROUP BY` query to work as expected.

And in the other PR https://github.com/growthbook/growthbook/pull/1050 I built the logic to fetch the table data from Mongo and if not available, query the datasource directly to get it.

Reply:
I've tried the following query (along with a number of variations) without success. Any guidance would be great.

```sql
SELECT
        table_name,
        "${projectId}" as table_catalog,
        table_schema,
        count(column_name)
 FROM
     ${projectId}.${dataset.schema_name}.INFORMATION_SCHEMA.COLUMNS
```
     
with a "GROUP BY table_name" clause tacked on the end.

Reply:
You have to group by any thing that isn't a scalar or summarized. So you may need `GROUP BY table_name, table_schema`

Reply:
@lukesonnet Thank you so much - I was able to get the `COUNT` and `GROUP BY` clauses working with this suggestion. Much appreciated!